grpc/src/main/java/io/stargate/grpc/service/interceptors/HeadersInterceptor.java

```diff
@@ -0,0 +1,30 @@
+package io.stargate.grpc.service.interceptors;
+
+import io.grpc.Context;
+import io.grpc.Contexts;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.stargate.grpc.service.Service;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HeadersInterceptor implements ServerInterceptor {
+  @Override
+  public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+      ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+    Map<String, String> stringHeaders = new HashMap<>();
+    for (String key : headers.keys()) {
+      if (key.endsWith("-bin")) {
```

Comment:
why do we need this edge case?

Reply:
gRPC allows both text and binary headers. Binary header keys always end with `-bin`.

`Connection.setCustomProperties` only accept text headers (and we know the tenant ID will come in a text header).

Reply:
> gRPC allows both text and binary headers
TIL